### DISCUSSION TITLE
Deprecation of space throwing

### DISCUSSION THREAD

**Discussion by ianyh** [ianyh/Amethyst#1682] [Announcements]
Due to changes Apple has made in macOS 15, Amethyst may no longer be able to support moving windows between native spaces. I don't currently know of a way outside of yabai's Dock injection to accomplish it, and I don't think that is the right way to go with this project. I understand and share the frustration. Amethyst is by no means going away, but I may need to deprecate this particular functionality.

I'm open to thoughts on how to approach the deprecation if anyone has feedback, or if anyone figures out a way to actually make it work, so I will leave comments open. I will be closing the relevant bugs, and redirecting them here.

**Comment by simonedamico** [comment]
ouch. It was my #1 reason for using amethyst

**Comment by mikalcallahan** [comment]
Hoping we can come up with an alternative way to move windows between spaces. Unlikely that Apple will expose an API for it but you never know

**Comment by miguelski** [comment]
i am using **BetterTouchTool** for now to accomplish this since the update.
i was hopping this can be fixed before my trial ends, but now i think i would need to purchase full version only for this feature

**Comment by rami3l** [comment]
Just to provide more context to this issue, [Aerospace](https://github.com/nikitabobko/AeroSpace) seems to have reinvented the "spaces" system themselves to overcome such restrictions... It's a disappointing move on Apple's side, really :(

**Comment by gkzhb** [comment]
There is a workaround https://github.com/ianyh/Amethyst/issues/1643#issuecomment-2133532570, and I managed to implement it with [Hammerspoon](https://www.hammerspoon.org/docs/index.html) (code commented below that issue https://github.com/ianyh/Amethyst/issues/1643#issuecomment-2424143589).
But the workaround may not work for every situations and may not be robust.

**Comment by rwiankowski** [comment]
Really sorry to read that! I've been enjoying Amethyst for a while now and heavily relied on the throwing feature as part of my productivity setup. I'm seriously considering downgrading to Sonoma.

**Comment by ianyh** [comment]
It seems like people are having some luck with other solutions. I might look into incorporating some of those techniques. I'm dreading upgrading my work laptop.

**Comment by bogdan-calapod** [comment]
Fwiw i switched to aerospace due to this. I can't say I love the fact that it doesn't use MacOS spaces but it works 🤷‍♂️

**Comment by ianyh** [comment]
I revived some old code I had for moving windows between spaces without private APIs in a build if anyone wants to try. https://github.com/ianyh/Amethyst/pull/1698

**Comment by ianyh** [comment]
Okay, I think I can make it work if people can enable the system shortcuts themselves—they aren't enabled by default and I can't find a way to manipulate them programmatically. They have default values that you can just enable, so I would only need more config info if they were changed from the default values.

**Reply by ianyh** [reply]
I can also track the starting space, so that would mean being able to go back to where you started after moving, but obviously that's horrifically slow.

**Reply by ianyh** [reply]
Actually, I take that back. I think I figured out how to look up the system shortcuts for switching between desktops, so I theoretically wouldn't need any configuration info, and I might be able to enable them directly.

**Reply by krvpal** [reply]
I have to admit I've been returning to this thread everyday in the hope there's a solution. I tried Aerospace but encountered some bugs and it doesn't do dynamic tiling. Really grateful for this project, it shows through when you realize the difference it has made to your workflow, and how disruptive it is when Apple pushes their breaking changes.

**Reply by rwiankowski** [reply]
Same here. Running Aerospace since the upgrade to Sequoia, and while I appreciate the creator's effort, I would rather go back to Amethyst.

**Comment by nkov** [comment]
@ianyh I looked at the updates from #1698 and seems that this has been fixed / merged, but I'm still experiencing the issue on Sequoia 15.7.7 and Amethyst 0.24.3 (129).

<img width="432" height="315" alt="image" src="https://github.com/user-attachments/assets/30d8e488-0874-4a70-986a-dfb45f264f54" />


I have these hotkeys set up and I'm using hyperkey for caps lock.

<img width="507" height="324" alt="image" src="https://github.com/user-attachments/assets/dd1797f1-7822-4a32-880a-7c203d500252" />
